packages/instantsearch.js/src/lib/server.ts

```diff
@@ -12,7 +12,8 @@ import type {
  * in `getServerState()`.
  */
 export function waitForResults(
-  search: InstantSearch
+  search: InstantSearch,
+  skipRecommend: boolean = false
```

Comment:
what's skipRecommend about? seems like we check hasRecommendWidget anyway?

Reply:
That's for `getServerState` for the pages router, as with dynamic widgets we *mount* the provided component a second time, we don't receive a response from recommend anymore as the results were cached.
Maybe the problem is that in the helper when there are no queries to be made it doesn't raise any event : https://github.com/algolia/instantsearch/blob/e55c345bce1969f4379db39de27642a7929c9411/packages/algoliasearch-helper/src/algoliasearch.helper.js#L1636-L1638
But I'm really not sure if it's worth changing now